test/controllers/EditEntryControllerSpec.scala

```diff
@@ -0,0 +1,168 @@
+import play.api.test.{FakeRequest, Injecting}
+import services.SiProtectedUserListService
+import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.gg.test.UnitSpec
+import uk.gov.hmrc.http.{ConflictException, NotFoundException}
+import uk.gov.hmrc.play.bootstrap.tools.Stubs
+import util.Generators
+import views.Views
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class EditEntryControllerSpec extends UnitSpec with Injecting with GuiceOneAppPerSuite with Generators with ScalaFutures with ScalaCheckDrivenPropertyChecks {
+  implicit val defaultPatience = PatienceConfig(timeout = Span(5, Seconds), interval = Span(5, Millis))
+
+  trait Setup {
+    val defaultSiProtectedUserConfig = siProtectedUserConfigGen.sample.get
+    val defaultAuthStrideEnrolmentsConfigGen = authStrideEnrolmentsConfigGen.sample.get
+    val appName = nonEmptyStringGen.sample.get
+    val stridePid = nonEmptyStringGen.sample.get
+
+    val mockSiProtectedUserListService = mock[SiProtectedUserListService]
+    val mockAuthConnector = mock[AuthConnector]
+    val views = inject[Views]
+    def editEntryController(siProtectedUserConfig: SiProtectedUserConfig = defaultSiProtectedUserConfig) = {
+      new EditEntryController(
+        siProtectedUserConfig,
+        mockSiProtectedUserListService,
+        views,
+        Stubs.stubMessagesControllerComponents(),
+        new StrideAction(mockAuthConnector, defaultAuthStrideEnrolmentsConfigGen, appName)
+      )
+    }
+    def expectStrideAuthenticated(): Unit = {
+      when(mockAuthConnector.authorise[Option[String]](any, any)(any, any)).thenReturn(Future.successful(Some(stridePid)))
+    }
+
+    def assertEditPageContainsFormFields(body: String): Unit = {
+      body should include("edit.entry.title")
+      body should include("edit.entry.header")
+      body should include("entry.form.action")
+      body should include("entry.form.nino")
+      body should include("entry.form.sautr")
+      body should include("entry.form.identityProvider")
+      body should include("entry.form.identityProviderId")
+      body should include("entry.form.addedByTeam")
+      body should include("edit.submit.button")
+      body should include("cancel.button")
+    }
+
+    def toEditRequestFields(entry: Entry): Seq[(String, String)] = {
+      Seq(
+        entry.entryId.map(e => "entryId" -> e),
+        Some("action" -> entry.action),
+        entry.nino.map(n => "nino" -> n),
+        entry.sautr.map(s => "sautr" -> s),
+        entry.identityProvider.map(s => "identityProvider" -> s),
+        entry.identityProviderId.map(s => "identityProviderId" -> s),
+        entry.group.map(s => "group" -> s),
+        entry.addedByTeam.map(s => "addedByTeam" -> s),
+        entry.updatedByTeam.map(s => "updatedByTeam" -> s)
+      ).flatten
+    }
+  }
+
+  "EditEntryController" should {
+    "forward to the edit entry view when GET /add is called" in new Setup {
```

Comment:
need to test scenarios for GET call, when service is shuttered, when the service.findEntry returns not found,  when the service returns an error and you display the internalServerError page

Reply:
May I assume responsibility for this? My ticket refactors the shuttering logic to be encapsulated with the `StrideAction`.

Reply:
resolved